packages/hoppscotch-sh-admin/src/modules/admin.ts

```diff
@@ -30,7 +26,14 @@ const getFirstTimeInfraSetupStatus = async () => {
 
 export default <HoppModule>{
   async onBeforeRouteChange(to, _from, next) {
-    const isAdmin = await getAdminStatus();
+    const res = await auth.getUserDetails();
+
+    // Allow performing the silent refresh flow for an invalid access token state
+    if (res.errors?.[0].message === UNAUTHORIZED) {
+      return next();
+    }
```

Comment:
Say the user is logged in with an invalid access token. We'd want to ensure the silent refresh flow happens while they attempt to navigate away. Here, we check to see if the query failure reason was unauthorized access (invalid access token being one) and we continue with the navigation in such a case. `@urql/auth-exchange` will ensure the silent refresh flow happens, [Ref](https://github.com/hoppscotch/hoppscotch/blob/main/packages/hoppscotch-sh-admin/src/main.ts#L41-L46).

Reply:
@jamesgeorge007 Oh okay. This makes sense.